### PR TITLE
Refactor server code

### PR DESCRIPTION
**Pull request by mj426382**

### CODE REVIEW

File: .github/workflows/container-push.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Login to Github Packages
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GHCR_PAT }}
+      
+      - name: Build image and push to Docker Hub and GitHub Container Registry
+        uses: docker/build-push-action@v2
+        with:
```
Comment (dmaretskyi) [inline .github/workflows/container-push.yml:28]: Does this build for ARM and Intel? do you need to specify arches in here?

File: .github/workflows/container-push.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          context: ./../
+          tags: |
+            ghcr.io/ethworks/hostlify:latest
+          push: ${{ github.ref == 'refs/heads/main' }}
```
Comment (dmaretskyi) [inline .github/workflows/container-push.yml:32]: Do you need this condition? I think this job runs only on main

File: .github/workflows/container-push.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to Github Packages
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GHCR_PAT }}
```
Comment (dmaretskyi) [inline .github/workflows/container-push.yml:24]: Do we need to set `GHCR_PAT` in secrets or would the default GITHUB_TOKEN work as well?

File: .github/workflows/container-push.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+name: Build image and Publish
```
Comment (dmaretskyi) [inline .github/workflows/container-push.yml:1]: Rename this file to `docker-build.yml`

File: src/app.ts [hunk 5]
```diff
@@ -1,86 +1,52 @@
...
-    process.exit(1)
-  }
-  await prepareServer()
-  console.log(`Server listening at ${address}`)
-})
+function runServer() {
+    server.listen(PORT, HOST, async(err, address) => {
+        if (err) {
+            throw new Error(err.message)
+    }
```
Comment (dmaretskyi) [inline src/app.ts:35]: formatting?

File: src/config.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+export interface Config {
+    domain: string | undefined
```
Comment (dmaretskyi) [inline src/config.ts:2]:
```suggestion
    domain?: string
```

File: src/nginx.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+import path from 'path'
+
+import { getConfig } from './config'
+import { NGINX_TEMPLATE, SERVER_TEMPLATE } from './nginxConfigTemplates'
+import { Files } from './types'
+
+const REPOSITORY_PATH = '/usr/share/nginx/html'
+const NIGINX_CONFIG_PATH= `/etc/nginx`
+
+async function createFilePathDirectoriesIfNecessary(filePath: string) {
```
Comment (dmaretskyi) [inline src/nginx.ts:14]: Not specific to nginx, extract to `./fs-utils.ts`

File: src/app.ts [hunk 8]
```diff
@@ -1,86 +1,51 @@
 import fastify from 'fastify'
```
Comment (dmaretskyi) [inline src/app.ts:1]: can you split this file into `api.ts` with startServer and `main.ts` that has the main function

File: src/types.ts [hunk 9]
```diff
@@ -1,11 +1,19 @@
...
+  mimetype: string
+  md5: string
+  mv: () => undefined
+}
+
+export interface Files {
+  [file: string]: File
+}
+
+export interface UploadParams {
```
Comment (dmaretskyi) [inline src/types.ts:17]: put those two right besides the api handler, or maybe inline them

File: src/nginx.ts [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+export async function updateApiConfig() {
+    const { domain } = getConfig()
+    if(!domain) {
+        throw new Error('Domain variable is not set!')
+    }
+
+    await createFilePathDirectoriesIfNecessary(`${NIGINX_CONFIG_PATH}/serverConfigs/config`)
+    const configFileContent = NGINX_TEMPLATE.replace('{{domain}}', domain)
+    await fs.writeFile(`${NIGINX_CONFIG_PATH}/nginx.conf`, configFileContent)
+    runNginx()
```
[the remark this answers is not in the document]
Comment (mj426382) [reply src/nginx.ts:57]: No, this only starts new process

File: .github/workflows/builder-check.yml [hunk 11]
```diff
@@ -15,10 +15,9 @@ jobs:
...
         id: truefipreview
-        uses: ethworks/hostlify-publish@v3.7
+        uses: ethworks/hostlify-publish@v3.25
         with:
           files: ./example
           server-url: truefipreview.io
-          owner: EthWorks
-          repo: hostlify
           access-token: ${{ secrets.GITHUB_TOKEN }}
+      - run: yarn lint
```
Comment (dmaretskyi) [inline .github/workflows/builder-check.yml:23]: prob deploy should be last

File: package.json [hunk 12]
```diff
@@ -12,13 +12,19 @@
...
-    "dev": "ts-node ./src/app.ts",
-    "clean": "rm -r build"
+    "start": "node ./build/main.js",
+    "dev": "ts-node ./src/main.ts",
+    "clean": "rm -r build",
+    "lint": "eslint './src/**/*.ts' --cache",
+    "lint:fix": "yarn lint --fix"
   },
   "publishConfig": {
     "@mj426382:registry": "https://npm.pkg.github.com"
```
Comment (dmaretskyi) [inline package.json:30]: is this needed?

File: src/types.ts (hunk 9 above)
Comment (mj426382) [reply src/types.ts:17]: It is used in nginx file, i don't think it is good idea, i remember out talk from yesterday, but i think it is the best option

File: package.json (hunk 12 above)
Comment (mj426382) [reply package.json:30]: dev is useful, we can remove clean

Comment (dmaretskyi) [reply package.json:29]: what about publish config? I think that's only for publishing NPM packages

Comment (mj426382) [reply package.json]: For future hah

Comment (mj426382) [reply package.json]: Deleted

**Pull request merged**